### DISCUSSION TITLE
Is DDoS protection part of Pro Plan

### DISCUSSION THREAD

**Discussion by geoff-impacttheory** [vercel/community#3108] [Feature Question] [answered]
### Question

Is the basic DDoS protection part of the Pro Plan, or is Enterprise needed?

**Comment by maddsua** [comment]
Actually, I had the same question. It all boils down to this: yes and no. Your app isn't gonna go down, but they'll still charge you a ton. Unless the support actually manages to answer to you and sort it out. I'd recommend using Cloudflare if you have concerns regarding DDoS attacks and such

**Comment by amyegan** [comment]
There is a longer description of Vercel's DDoS mitigation at [vercel.com/docs/security/ddos-mitigation](https://vercel.com/docs/security/ddos-mitigation), but here are some highlights:

> Different attack types can target different layers of the [OSI model](https://en.wikipedia.org/wiki/OSI_model).
> 
> DDoS attacks often target the [network](https://vercel.com/docs/security/ddos-mitigation#layer-3-ddos) (layer 3) and [transport](https://vercel.com/docs/security/ddos-mitigation#layer-4-ddos) (layer 4) layers of the OSI model. Vercel mitigates against these attacks, and protects the entire platform and all customers from attacks that would otherwise affect reliability.


> [Vercel helps to mitigate against L3 and L4 DDoS attacks](https://vercel.com/docs/security/ddos-mitigation#open-system-interconnection-osi-model) at the platform level. Usage will be incurred for requests that are successfully served prior to us automatically mitigating the event. Mitigation usually takes place within one minute.
> 
> Usage will be incurred for requests that are not recognized as a DDoS event, such as bot and crawler traffic.
>
> You should [monitor your usage](https://vercel.com/docs/concepts/get-started/monitor) and utilize [Edge Middleware](https://vercel.com/docs/concepts/functions/edge-middleware) to protect against undesired traffic based on its IP, User-Agent header value, or other identifiers.

**Comment by yangli-io** [comment]
My app goes down everytime theres a ddos attack. Usually the attacks doesnt last long because we have cloudflare mitigations but Vercel doesnt handle it nicely. I wish vercel just charged me, however all it does is just kill my app for like 10 minutes. So what happens is an attack attempts to attack and some traffic manages to reach the Vercel server. Vercel decides it's too much and proceeds to shut down our app for 5-10mins. It's terrible...

**Comment by leerob** [comment, marked as answer]
Hey, wanted to follow up here.

We just launched **[Attack Challenge Mode](https://vercel.com/changelog/prevent-malicious-traffic-with-attack-challenge-mode-for-vercel-firewall)** –an extra layer of defense for DDoS and malicious traffic, free for all plans. When enabled, the Vercel Firewall will serve a challenge page to help mitigate attacks.

Hope this helps!
